staging/src/k8s.io/kubectl/pkg/cmd/util/podcmd/podcmd.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podcmd
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// FindContainerByName selects the named container from the spec of
+// the provided pod or return nil if no such container exists.
+func FindContainerByName(pod *v1.Pod, name string) *v1.Container {
```

Comment:
We have identical method in https://github.com/kubernetes/kubernetes/blob/076bd6c4010d9573d89174a1ddb17a35018c2db0/staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go#L197

Bonus points if we could re-use https://github.com/kubernetes/kubernetes/blob/076bd6c4010d9573d89174a1ddb17a35018c2db0/staging/src/k8s.io/kubectl/pkg/util/podutils/podutils.go#L30
for defaulting the container as described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/2227-kubectl-default-container/README.md

Currently that's being used in https://github.com/kubernetes/kubernetes/blob/076bd6c4010d9573d89174a1ddb17a35018c2db0/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go#L752 which deserves squashing with your approach as well.

Reply:
Yeah I want to collapse these because selecting the default container should be 100% standard.